programs/ssl/ssl_client2.c

```diff
@@ -3160,10 +3171,25 @@ int main( int argc, char *argv[] )
     mbedtls_printf( "  . Closing the connection..." );
     fflush( stdout );
 
-    /* No error checking, the connection might be closed already */
-    do ret = mbedtls_ssl_close_notify( &ssl );
-    while( ret == MBEDTLS_ERR_SSL_WANT_WRITE );
-    ret = 0;
+    /*
+     * Most of the time sending a close_notify before closing is the right
+     * thing to do. However, when the server already knows how many messages
+     * are expected and closes the connection by itself, this alert becomes
+     * redundant. Sometimes with DTLS this redundancy becomes a problem by
+     * leading to a race condition where the server might close the connection
+     * before seeing the alert, and since UDP is connection-less when the
+     * alert arrives is wil be seen as a new connection, which will fail as
```

Comment:
Trivial - You don't have to fix this but:
> "alert arrives is wil be seen as a new connection"
Should be:
> "alert arrives it will be seen as a new connection"